lib/ansible/utils/collection_loader/_collection_finder.py

```diff
@@ -531,6 +567,42 @@ def _validate_final(self):
             # only search within the first collection we found
             self._subpackage_search_paths = [self._subpackage_search_paths[0]]
 
+    def exec_module(self, module):
+        if not _meta_yml_to_dict:
```

Comment:
Since this duplicates much of `load_module`, maybe we could just call the 3 step loader from `load_module` so that we eliminate the duplication?

Reply:
I refactored those two methods to both use a new method `_load_module`, similar to what I did for find_module/find_spec in the commit before it.